include/CL/sycl/platform/detail/platform_tail.hpp

```diff
@@ -18,32 +18,35 @@ namespace sycl {
     @{
 */
 
-/** Return a list of all available devices
+/** Returns a vector class containing all SYCL devices associated with
+    this SYCL platform.
 
     Return synchronous errors via SYCL exception classes.
 */
 vector_class<device>
 platform::get_devices(info::device_type device_type) const {
-  // Start with the default device
-  vector_class<device> devices = { {} };
+  /** If \c get_devices is called with the host platform,
+      returns the host_device.
+  */
+  if(is_host()) return { {} };
```

Comment:
Actually I have the feeling that you are doing an explicit manual dispatch here where virtual inheritance would be more logical.
So why not having `platform::get_devices` virtual and moving this into `detail::host_platform::get_devices`?